### ISSUE TITLE
get_result() waits forever issue

### ISSUE THREAD

**Issue by omerb01** [pywren/pywren-ibm-cloud#246]
as discussed in #209 (just to document this issue) and regarding #144 .
we need to improve `get_result()` so it won't wait forever in any case.

I think that when a single action is killed and `get_result()` waits more than 10mins (max run time configured for actions inside CF), we can assume that it got OOM error and terminate PyWren accordingly.

Example code that should raise OOM error after 10mins: (now it runs forever)
```python
import pywren_ibm_cloud as pywren

def my_func(args):
    mem = [0] * 1024 ** 3  # 1GB
    return "Hello World!"

pw = pywren.ibm_cf_executor()
pw.map(my_func, range(2))
print(pw.get_result())
```

**Comment by omerb01** [comment]
in addition, I found that `get_result()` doesn't detect timeout errors when actions run more than 10min, this should be thrown regardless of OOM errors (if possible)
if it isn't possible to differ timeout errors and OOM errors, I suggest throwing timeout error in both cases.

Example code that should throw "timeout" error but actually waits forever:
```python
import pywren_ibm_cloud as pywren
import time

def my_func(args):
    time.sleep(1200)  # 20mins
    return "Hello World!"

pw = pywren.ibm_cf_executor()
pw.map(my_func, range(2))
print(pw.get_result())
```

**Comment by JosepSampe** [comment]
1st post and 2nd pot are different issues. #248 fixes issue from 2nd post

**Comment by omerb01** [comment]
@JosepSampe thanks, tested it and 2nd post was solved.

**Comment by JosepSampe** [comment]
I added a fix in #253 in order to prevent get_result() wait forever for results (1st post).  Note that currently this patch/implementation only works for RabbitMQ. So we can start to test it when merged and suggest improvements (if any). Will summit a patch for object Storage in another PR.

**Comment by omerb01** [comment]
@JosepSampe Thanks!
